### DISCUSSION TITLE
why chaining order of nullish is optional().nullable()?

### DISCUSSION THREAD

**Discussion by p10ns11y** [colinhacks/zod#1565] [Q&A] [answered]
I would like to know the reasoning behind the decision why `nullish`'s chaining order is `optional().nullable()`?

I think if the order is `nullable().optional()` then it would play nicely with `required`

Use case:

```js
const schema = z.object({
  nullishProp: z.string().nullish(), // optional().nullable()
  anotherProp: z.number(),
})
```
The latest code will not make `nullishProp` as required prop  when we do `schema.required({ nullishProp: true })`

If we had `nullable().optional()` for `nullish` then schema.required({ nullishProp: true }) make `nullishProp` required and allow `null` value as well

**Comment by JacobWeisenburger** [comment, marked as answer]
This is a very interesting thing that you have found. I'm not sure if this is intended behavior or not. Feel free to submit a PR for this.
